### PR TITLE
SEO for Heroku

### PR DESCRIPTION
**Pull request by ajeetraina**

### CODE REVIEW

**Comment by ajeetraina** [comment]
@simonprickett Please wait till I complete this activity.

File: docs/create/from-source/index-from-source.mdx [hunk 1]
```diff
@@ -7,6 +7,12 @@ authors: [ajeet]
 ---
 
 ### Step 1: Download, extract and compile Redis
+
+Redis stands for  REmote  DIctionary  Server. Redis is an open source, in-memory, key-value data store most commonly used as a primary database, cache, message broker, and queue. Redis cache delivers sub-millisecond response times, enabling fast and powerful real-time applications in industries such as gaming, fintech, ad-tech, social media, healthcare, and IoT.
+
+In order to install Redis from source, first you need to download the latest Redis source code. 
+The Redis source code files are available on [this download](https://download.redis.io/redis-stable.tar.gz). You can verify the integrity of these downloads by checking them against the digests in the [redis-hashes git repository](https://github.com/redis/redis-hashes)
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
The Redis source code is available to download [here](https://download.redis.io/redis-stable.tar.gz). You can verify the integrity of these downloads by checking them against the digests in the [redis-hashes git repository](https://github.com/redis/redis-hashes)
```

File: docs/create/from-source/index-from-source.mdx [hunk 2]
```diff
@@ -7,6 +7,12 @@ authors: [ajeet]
...
+
  
 ```
 wget http://download.redis.io/redis-stable.tar.gz
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
wget https://download.redis.io/redis-stable.tar.gz
```

File: docs/create/from-source/index-from-source.mdx [hunk 3]
```diff
@@ -27,13 +33,18 @@ The binaries that are now compiled are available in the src directory.
 
 
 ### Step 2: Running Redis Server
+
+Install Redis server by running the following redis command:
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
Install the Redis server by running the following command:
```

File: docs/create/from-source/index-from-source.mdx [hunk 4]
```diff
@@ -27,13 +33,18 @@ The binaries that are now compiled are available in the src directory.
...
  
 ```
 $ redis-server
 ```
+Please note that you don't need to restart the Redis service
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
Please note that you don't need to restart the Redis service.
```

File: docs/create/from-source/index-from-source.mdx [hunk 5]
```diff
@@ -27,13 +33,18 @@ The binaries that are now compiled are available in the src directory.
...
 
 ### Step 3: Interacting with Redis Client
 
+Once the redis installation gets completed, you can use redis client software to perform the remote connection to the redis instance.
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
Once the Redis installation has completed, you can use the Redis client to connect to the Redis server.
```

File: docs/create/from-source/index-from-source.mdx [hunk 6]
```diff
@@ -27,13 +33,18 @@ The binaries that are now compiled are available in the src directory.
...
+Use the below redis cli command to SET and GET the data values:
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
Use the following commands to store and retrieve a string:
```

File: docs/create/from-source/index-from-source.mdx [hunk 7]
```diff
@@ -43,9 +54,12 @@ redis> get foo
 ```
 
 Please note: By default, Redis has 0-15 indexes for databases, you can change that number databases NUMBER in redis.conf.
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
```

Removing as we discourage use of this and the `SELECT` command and Redis Enterprise doesn'thave it.

File: docs/create/from-source/index-from-source.mdx [hunk 8]
```diff
@@ -43,9 +54,12 @@ redis> get foo
...
+The redis.conf is a configuration file recommended in case you want to configure Redis.
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
`redis.conf` is the Redis configuration file, used to configure the behavior of the Redis Server.  For more information on the available configuration options, check out the [documentation on redis.io](https://redis.io/docs/manual/config/).
```

File: docs/create/from-source/index-from-source.mdx [hunk 9]
```diff
@@ -43,9 +54,12 @@ redis> get foo
...
+
 
 ### Next Step
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
### Next Steps
```

File: docs/create/from-source/index-from-source.mdx [hunk 10]
```diff
@@ -43,9 +54,12 @@ redis> get foo
...
 
 - [Connect to Redis database using RedisInsight](/explore/redisinsight)
```
Comment (simonprickett) [inline docs/create/from-source/index-from-source.mdx]:
```suggestion
- [Connect to a Redis database using RedisInsight](/explore/redisinsight)
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 11]
```diff
@@ -16,14 +16,15 @@ Here are few popular terminologies used in Heroku:
 - Dynos: The Heroku Platform uses the container model to run and scale all Heroku apps. The containers used at Heroku are called “dynos". Dynos are isolated, virtualized Linux containers that are designed to execute code based on a user-specified command.
 - Buildpack: It is a config script for the build automation process, describing how a container image should be created.
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
- Buildpack: This is a config script for the build automation process, describing how a container image should be created.
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 12]
```diff
@@ -16,14 +16,15 @@ Here are few popular terminologies used in Heroku:
...
 - Add-ons: These are tools and services for extending a Heroku application's functionality, such as data storage and processing, monitoring, or analytics.
-- Heroku CLI: It is a tool for building and running Heroku apps from within the terminal. (Docker, too, uses its own CLI for working with the platform.)
+- Heroku CLI: It is a tool for building and running Heroku app from within the terminal. (Docker, too, uses its own CLI for working with the platform.) Learn about [Heroku Dev Center](https://devcenter.heroku.com/)
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
- Heroku CLI: This is a tool for building and running Heroku apps from within the terminal. (Docker, too, uses its own CLI for working with the platform.) Learn about [Heroku Dev Center](https://devcenter.heroku.com/)
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 13]
```diff
@@ -16,14 +16,15 @@ Here are few popular terminologies used in Heroku:
...
 - Git: It is a popular version control system for tracking changes to a software's source code. Heroku makes it easy to manage your app deployments with git, and has built-in integrations with the GitHub hosting platform for git repositories.
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
- Git: A popular version control system for tracking changes to a software's source code. Heroku makes it easy to manage your app deployments with git, and has built-in integrations with the GitHub hosting platform for git repositories.
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 14]
```diff
@@ -16,14 +16,15 @@ Here are few popular terminologies used in Heroku:
...
 
 Heroku recognizes an app as being written in Go by the existence of a `go.mod` file in the root directory. Heroku also supports govendor, godep & GB, but this tutorial focuses only on Go modules. Here's a quickstart guide to deploy Go apps on Heroku using Redis. We will be deploying a sample Chat application written in Go.
 
 ### Step 1. Create a Redis Enterprise Cloud Database
 
 Create your free Redis Enterprise Cloud account by visiting [this link](https://redis.com/try-free).
+Creating a heroku account is free of cost.
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
Creating a Heroku account is free of charge.
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 15]
```diff
@@ -39,19 +40,24 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
 
 ### Step 2. Create a Heroku account
 
-If you are using Heroku for the first time, create your new Heroku account [through this link](https://signup.heroku.com/login).
+If you are using Heroku for the first time, visit Heroku website and create your new Heroku account [through this link](https://signup.heroku.com/login).
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
If you are using Heroku for the first time, visit the Heroku website and create your new Heroku account [through this link](https://signup.heroku.com/login).
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 16]
```diff
@@ -39,19 +40,24 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
...
 
 ![heroku](create_heroku.png)
 
 
 ### Step 3. Install the Heroku CLI on your system
 
+The Heroku Command Line Interface (CLI) lets you create and manage Heroku apps directly from the terminal. It's an essential part of using Heroku. In order to install Heroku CLI, run the below command:
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
The Heroku Command Line Interface (CLI) lets you create and manage Heroku apps directly from the terminal. It's an essential part of using Heroku. In order to install Heroku CLI, run the following command:
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 17]
```diff
@@ -39,19 +40,24 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
...
+
    ```macos
     brew install heroku
    ```
 
 ### Step 4. Login to Heroku
 
+Use the following Heroku Command Line interface to login to Heroku dashboard:
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
Use the following Heroku Command Line commands to login to Heroku dashboard:
```

File: docs/create/heroku/herokugo/index-herokugo.mdx [hunk 18]
```diff
@@ -64,14 +70,16 @@ If you are using Heroku for the first time, create your new Heroku account [thro
...
+
+First you will need to clone the GitHub repository to configure it as a local repository.
 
    ```bash
     git clone https://github.com/redis-developer/basic-redis-chat-demo-go
    ```
 
 
-
+The `basic-redis-chat-demo-go` is the name of the project directory.
```
Comment (simonprickett) [inline docs/create/heroku/herokugo/index-herokugo.mdx]:
```suggestion
`basic-redis-chat-demo-go` is the name of the project directory.
```

File: docs/create/heroku/herokupython/index-herokupython.mdx [hunk 19]
```diff
@@ -33,15 +35,21 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
 
 [Follow this link to create a Redis Enterprise Cloud](/create/rediscloud) subscription and database. Once you create the database, you will be provisioned with a unique database endpoint URL, port and password. Save these for future reference.
 
+You can use redis client command line to quickly verify the connection to the Redis instance URL and access the database.
```
Comment (simonprickett) [inline docs/create/heroku/herokupython/index-herokupython.mdx]:
```suggestion
You can use the Redis CLI to quickly verify the connection to the Redis instance URL and access the database.
```

File: docs/create/heroku/herokupython/index-herokupython.mdx [hunk 20]
```diff
@@ -33,15 +35,21 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
...
+
 ### Step 2. Create a Heroku account
 
-If you are using Heroku for the first time, create your new Heroku account [through this link](https://signup.heroku.com/login).
+If you are using Heroku platform for the first time, create your new Heroku account [through this link](https://signup.heroku.com/login).
```
Comment (simonprickett) [inline docs/create/heroku/herokupython/index-herokupython.mdx]:
```suggestion
If you are using the Heroku platform for the first time, create your new Heroku account [through this link](https://signup.heroku.com/login).
```

File: docs/create/heroku/herokupython/index-herokupython.mdx [hunk 21]
```diff
@@ -33,15 +35,21 @@ For a limited time, use **TIGER200** to get **$200** credits on Redis Enterprise
...
+You can refer to [Heroku documentation](https://devcenter.heroku.com/categories/reference)
 
 ![heroku](create_heroku.png)
 
 
 ### Step 3. Install Heroku CLI on your system
 
+Run the following command to install heroku in your system.
```
Comment (simonprickett) [inline docs/create/heroku/herokupython/index-herokupython.mdx]:
```suggestion
Run the following command to install the Heroku CLI on your system.
```

File: docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx [hunk 22]
```diff
@@ -139,7 +139,7 @@ If your app doesn point to the right repository, you can manually add it:
 ```
 
 
-### Step 8. Accessing the application
+### Step 8. Accessing the heroku app
```
Comment (simonprickett) [inline docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx]:
```suggestion
### Step 8. Accessing the Heroku app
```

File: docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx [hunk 23]
```diff
@@ -139,7 +139,7 @@ If your app doesn point to the right repository, you can manually add it:
...
 
 Open [https://powerful-fortress-83061.herokuapp.com/](https://powerful-fortress-83061.herokuapp.com/) to see your application.
```
Comment (simonprickett) [inline docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx:144]: This proably shouldn't be a link as this won't be their application's URL.

File: docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx [hunk 24]
```diff
@@ -169,7 +169,7 @@ The application uses cookies to identify users.  On the first request, the user
 - Get requests number for user: `GET requests.<USER_IDENTIFIER>`
   - E.g `GET requests.0cbc6611f5540bd0809a388dc95a615b`
 
-### Code for rate limiting
+### Code for rate limit
```
Comment (simonprickett) [inline docs/create/heroku/ratelimiting-go/index-ratelimitinggo.mdx]:
```suggestion
### Code for Rate Limiting
```

Changed this back to limiting as limit didn't make sense here.

File: docs/howtos/security/tls.mdx [hunk 25]
```diff
@@ -35,7 +35,23 @@ In this article, we will focus on the Two-Way SSL, and using Redis Enterprise.
 * [Python](/develop/python/), [Node](/develop/node), and [Java](/develop/java) installed if you want to test various languages.
 
 
-### Simple  Test
+## Simple  Test
+
+## Step 1. Run a Redis server
+
+You can either run Redis server in a Docker container or directly on your machine. Follow the below command to setup a Redis server locally on Mac OS:
```
Comment (simonprickett) [inline docs/howtos/security/tls.mdx]:
```suggestion
You can either run Redis server in a Docker container or directly on your machine. Use the following commands to setup a Redis server locally on Mac OS:
```

File: docs/howtos/security/tls.mdx [hunk 26]
```diff
@@ -60,7 +76,7 @@ In this tutorial, we will be using Docker to copy the certificate.
 docker cp redis-node1:/etc/opt/redislabs/proxy_cert.pem ./certificates
 ```
 
-## Step 2. Generate a New Client Certificate
+## Step 3. Generate a New Client Certificate
 
 Using the Two-Way SSL, you need to have a certificate for the client that will be used by Redis database proxy to trust the client.In this tutorial, we will use a self-signed certificate using OpenSSL. We will be creating a certificate for an application named `app_001`. Please note that you can create as many certificates as you want, or reuse this one for all servers/applications.
```
Comment (simonprickett) [inline docs/howtos/security/tls.mdx]:
```suggestion
Using the Two-Way SSL, you need to have a certificate for the client that will be used by Redis database proxy to trust the client. In this tutorial, we will use a self-signed certificate using OpenSSL. We will be creating a certificate for an application named `app_001`. Please note that you can create as many certificates as you want, or reuse this one for all servers/applications.
```

File: docs/howtos/security/tls.mdx [hunk 27]
```diff
@@ -81,7 +97,7 @@ openssl req \
 This command generate a new client key (`client_key_001.pem`) and certificate (`client_cert_001.pem`) with no passphrase.
```
Comment (simonprickett) [inline docs/howtos/security/tls.mdx]:
```suggestion
This command generates a new client key (`client_key_001.pem`) and certificate (`client_cert_001.pem`) with no passphrase.
```

**Comment by ajeetraina** [comment]
@simonprickett This was on my pending to-do list. I am happy that I was able to complete this effort.

**Comment by ajeetraina** [comment]
Please go ahead and merge it. All the Best.

**Pull request merged**